src/screen/dashboard/pane.rs

```diff
@@ -22,6 +21,7 @@ pub enum Message {
     Popout,
     Merge,
     ScrollToBottom,
+    MaintainFocus,
```

Comment:
I think `MaintainFocus` can be removed from here and `sidebar` and we can instead expose `LeftClick` from `event.rs` which is always emitted (even when event is captured) and use that to always call `refocus` in dashboard. 

This is logically simple to think about. Any time the user left clicks within the application, doesn't matter where, we refocus our text input. Clicking doesn't happen and isn't a high volume event so this isn't a big deal and then we don't have to figure out all places someone can click to lose focus from text input.

This actually probably allows us to remove almost most code in this PR that isn't focus history (which we still need obviously). Everything else should be captured by the above.

Reply:
The only time we wouldn't refocus is if `self.focus` is None if we unfocused the pane entirely

Reply:
If we do the above, we should be fully covered w/ input focus via left click anywehre when pane has focus, window focus changing, and clicking on a pane from unfocused state.